packages/events/src/events.ts

```diff
@@ -0,0 +1,264 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/listener
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+// tslint:disable-next-line:no-any
+export type Event = any;
+
+/**
+ * Async event listener function
+ */
+export type ListenerFunction = (
```

Comment:
I think most listeners won't be interested in the `eventType` value. Can we switch the order of the arguments to allow consumers of this API to ignore `eventType`?

```ts
export type ListenerFunction = (
  event: Event,
  eventType: string
) => Promise<void>;
```

A test to verify this works:
```ts
registry.subscribe(source, 'starting', data => {/* do something with data */});
```

Also, have you considered how to enable type checking for events? I believe TypeScript has a way how to express that certain `eventType` values are linked to certain `event` data formats.

For example, I can imagine using the same trick we do for `BindingKeys`.

```ts
// a simplified mock-up
subscribe<T>(source, eventType: EventType<T>, listenerFunction: EventListener<T>) {
  // ...
}
```